portfolio/src/main/webapp/script.js

```diff
@@ -35,50 +36,91 @@ function toggleCourseList(listObject) {
+// function that changes slideshow's picture to the next
 // picture when the user presses the next button
 function next() {
-  clearInterval(myTimer);
+  if (!paused) {
+    clearInterval(myTimer);
+  }
   currImageNum++;
   if (currImageNum === 17) {
     currImageNum = 1;
   }
   const newImgPath = loc + currImageNum + ".jpg";
   document.getElementById("galleryimg").src = newImgPath;
-  myTimer = setInterval(loopOverImages, sliderSpeed);
+  if (!paused) {
+    myTimer = setInterval(loopOverImages, sliderSpeed);
+  }
 }
 
-// function that changes slideshow's picture to the previous 
+// function that changes slideshow's picture to the previous
 // picture when the user presses the previous button
 function prev() {
-  clearInterval(myTimer);
+  if (!paused) {
+    clearInterval(myTimer);
+  }
   currImageNum--;
   if (currImageNum === 0) {
     currImageNum = 16;
   }
   const newImgPath = loc + currImageNum + ".jpg";
   document.getElementById("galleryimg").src = newImgPath;
-  myTimer = setInterval(loopOverImages, sliderSpeed);
+  if (!paused) {
+    myTimer = setInterval(loopOverImages, sliderSpeed);
+  }
 }
 
 // function that is called every sliderSpeed seconds
-// changes slideshow to display next image in list of 
+// changes slideshow to display next image in list of
 // images
 function loopOverImages() {
-  currImageNum++;
-  if (currImageNum === 17) {
-    currImageNum = 1;
+  if (!paused) {
+    currImageNum++;
+    if (currImageNum === 17) {
+      currImageNum = 1;
+    }
+    const newImgPath = loc + currImageNum + ".jpg";
+    document.getElementById("galleryimg").src = newImgPath;
   }
-  const newImgPath = loc + currImageNum + ".jpg";
-  document.getElementById("galleryimg").src = newImgPath;
 }
 
-
 // modifies slider speed if user changes value
 // of slider on gallery page
 function sliderMoved() {
-  clearInterval(myTimer);
+  if (!paused) {
+    clearInterval(myTimer);
+  }
   const sliderValue = document.getElementById("galleryslider").value;
   sliderSpeed = sliderValue * 1000;
-  myTimer = setInterval(loopOverImages, sliderSpeed);
+  if (!paused) {
+    myTimer = setInterval(loopOverImages, sliderSpeed);
+  }
+}
+
+
+//function to pause or resume slideshow when user clicks
+//on that window
+function togglePause() {
+  if (paused === true) {
+    const statusImg = document.getElementById("pauseplay")
+    statusImg.src = "/images/play.png";
+    statusImg.style.display = "block";
+    window.setTimeout(function () {
+      $("#pauseplay").fadeOut();
+      statusImg.style.display = "none";
+    }, 250);
+    myTimer = setInterval(loopOverImages, sliderSpeed);
+    paused = false;
+  } else {
+    console.log("Hi");
+    clearInterval(myTimer);
+    const statusImg = document.getElementById("pauseplay")
+    statusImg.src = "/images/pause.png";
+    statusImg.style.display = "block";
+    window.setTimeout(function () {
+      $("#pauseplay").fadeOut();
+      statusImg.style.display = "none";
+    }, 500);
```

Comment:
Is there a reason that the play fadeout time and pause fadeout time are different?

Reply:
Accidental, sorry.